contracts/src/Teleporter/upgrades/TeleporterUpgradeable.sol

```diff
@@ -43,20 +48,19 @@ abstract contract TeleporterUpgradeable {
         );
 
         teleporterRegistry = TeleporterRegistry(teleporterRegistryAddress);
-        _minTeleporterVersion = teleporterRegistry.getLatestVersion();
-    }
-
-    /**
-     * @dev Updates `_minTeleporterVersion` to the latest version.
-     */
-    function updateMinTeleporterVersion() external {
-        _minTeleporterVersion = teleporterRegistry.getLatestVersion();
+        minTeleporterVersion = teleporterRegistry.getLatestVersion();
     }
 
     /**
-     * @dev Gets the minimum Teleporter version that is allowed to call this contract.
+     * @dev This is a virtual function that should be overridden to update the `minTeleporterVersion`
+     * allowed for modifier `onlyAllowedTeleporter`, and emit {MinTeleporterVersionUpdated} event after.
+     * The derived contract should call this function after there is a new Teleporter version
+     * registered in the `TeleporterRegistry`, and the derived contract no longer needs to receive
+     * messages from the previous Teleporter versions.
+     *
+     * Note: To prevent anyone from being able to call this function, which would disallow messages
+     * from old Teleporter versions from being received, this function should be safeguarded with access
+     * controls. For example, if the derived contract has an owner/admin, only they can call this function.
```

Comment:
I don't think we should include so much implementation detail here. I'd say the important pieces to note are that this should update the `minTeleporterVersion` used in the modifier, and that it should use the `TeleporterRegistry` to do so. Details such as emitting events and access controls are best documented in the README and example apps.

Reply:
I think it's still good to point out that to protect this call from being called by anyone it should be implemented by access controls. Agreed to remove the part in the comment about how this function should be called after a new version is registered in TeleporterRegistry, etc, as this is written in README.

Reply:
Sounds good to me 👍